pkg/controller/mongodb/replica_set_controller.go

```diff
@@ -468,6 +478,17 @@ func (r ReplicaSetReconciler) buildAutomationConfigConfigMap(mdb mdbv1.MongoDB)
 		Build(), nil
 }
 
+// getMongodConfigModification will merge the additional configuration in the CRD
+// into the configuration set up by the operator.
+func getMongodConfigModification(mdb mdbv1.MongoDB) automationconfig.Modification {
+	return func(ac *automationconfig.AutomationConfig) {
+		for i, _ := range ac.Processes {
+			// Mergo requires both objects to have the same type
+			mergo.Merge(&ac.Processes[i].Args26, objx.New(mdb.Spec.AdditionalMongodConfig.Object), mergo.WithOverride)
```

Comment:
`objx` has a built-in merge function but unfortunately it does not support recursive merging. As we are already using `mergo` it seemed like a good fit here as well.

Reply:
`mergo.WithOverride` will do a direct override. I think this means if a user provides any tls settings, we lose all of the operator configured ones. I would have to double check that though! (maybe we could add a unit test for this)

Reply:
I checked and it seems as if mergo is smart enough to not do this. I also added another unit test to ensure that this is the case.